includes/amp-post-types-support.php

```diff
@@ -0,0 +1,35 @@
+<?php
+/**
+ * AMP Post types support.
+ *
+ * @package AMP
+ * @since 0.6
+ */
+
+/**
+ * Declare core post types support.
+ *
+ * @since 0.6
+ */
+function amp_core_post_types_support() {
+	add_post_type_support( 'post', AMP_QUERY_VAR );
+}
+add_action( 'init', 'amp_core_post_types_support' );
+
+/**
+ * Declare custom post types support.
+ *
+ * This function should only be invoked through the 'after_setup_theme' action to
+ * allow plugins/theme to overwrite the post types support.
+ *
+ * @since 0.6
+ */
+function amp_custom_post_types_support() {
+	// Listen to post types settings.
+	foreach ( AMP_Settings_Post_Types::get_instance()->get_settings() as $post_type => $enabled ) {
+		if ( true === $enabled ) {
+			add_post_type_support( $post_type, AMP_QUERY_VAR );
+		}
+	}
+}
+add_action( 'after_setup_theme', 'amp_custom_post_types_support' );
```

Comment:
I wonder if this shouldn't have a lower priority, like `5`. 

The _s starter theme and Twenty Seventeen among others do their `add_theme_support` calls in `after_setup_theme` at priority 10. 
https://github.com/Automattic/_s/blob/efd37d13a6622bcd20e8c7b35309102c06981c71/functions.php#L10-L84
https://github.com/WordPress/wordpress-develop/blob/a825a181e11b1a89e76d79d22bd2c220f45b3741/src/wp-content/themes/twentyseventeen/functions.php#L20-L217

So if they also do calls to `add_post_type_support()` I'd expect them to do it here as well, for example: https://github.com/ryelle/Anadama-React/blob/21c8f550bc3ea6bbb7407891bf06845316b0ebf6/functions.php#L78-L82